frontend/src/routes/Applications/Overview.tsx

```diff
@@ -871,46 +893,6 @@ export default function ApplicationsOverview() {
                 }
                 rowActionResolver={rowActionResolver}
             />
-        )
-    }
-
-    useEffect(() => {
-        const canCreateApplicationPromise = canUser('create', ApplicationDefinition)
-        canCreateApplicationPromise.promise
-            .then((result) => setCanCreateApplication(result.status?.allowed!))
-            .catch((err) => console.error(err))
-        return () => canCreateApplicationPromise.abort()
-    }, [])
-    useEffect(() => {
-        const canDeleteApplicationPromise = canUser('delete', ApplicationDefinition)
-        canDeleteApplicationPromise.promise
-            .then((result) => setCanDeleteApplication(result.status?.allowed!))
-            .catch((err) => console.error(err))
-        return () => canDeleteApplicationPromise.abort()
-    }, [])
-    useEffect(() => {
-        const canDeleteApplicationSetPromise = canUser('delete', ApplicationSetDefinition)
-        canDeleteApplicationSetPromise.promise
-            .then((result) => setCanDeleteApplicationSet(result.status?.allowed!))
-            .catch((err) => console.error(err))
-        return () => canDeleteApplicationSetPromise.abort()
-    }, [])
-
-    useEffect(() => {
-        setTableContent(generateTable())
-
-        const interval = setInterval(() => {
-            setTableContent(generateTable())
-        }, 5000)
-        return () => {
-            clearInterval(interval)
-        }
-    }, [canCreateApplication, canDeleteApplication, canDeleteApplicationSet, data])
```

Comment:
Because this call to `useEffect` has `data` as a dependency, each time the search query updates the data (every 5 seconds) the effect will fire again, so we get an updated rendering every 5 seconds. But if a change to a local kube resources comes through via the watch and recoil state, this has the effect of delaying that update until one of the periodic search query refreshes. Removing this effect, I see a rendering every 5 seconds, plus immediate rendering when recoil state is updated. I did not observe excessive rerendering.